config/api/routes/lock.php

```diff
@@ -9,44 +9,70 @@
         'pattern' => '(:all)/lock',
         'method'  => 'GET',
         'action'  => function (string $path) {
-            return $this->parent($path)->lock()->get();
+            if ($lock = $this->parent($path)->lock()) {
+                return $lock->get();
+            }
+
+            return [
+                'locked' => false
```

Comment:
I'm not 100 % sure about the API response changes (this comment applies to all changes in this file):

It would probably be cleaner to respond to all requests with `HTTP 501 Not implemented` and to implement the fallback logic in the frontend. It's a bit more complex, but would save us trouble for edge-cases in API usage where the API client needs to know if locking actually worked or is not supported at all.

Because with the current new implementation, clients that use the locking functionality would think that all requests went through fine and would signal that to the user.

Reply:
I think any Exception thrown in the API (which would be needed to pass the specific 501 code) will result in console errors. I don't think there should be a console error every 5 seconds when actually everything is working as intended.

Reply:
Hm, that's true, but I think we should first think about how to ensure a clean API structure and after that about how to use the API in the frontend.

A solution could be the following:

- The `GET <model>/lock` request could return:
  ```json
  {
    "locked": false,
    "lockingSupported": false
  }
  ```
  If supported, it would of course return `lockingSupported: true`. In both cases the HTTP code would be 200 as the `GET` was of course successful.
- Based on that value, the client could decide whether to run the locking code for that model. If the `GET` returned that locking is supported, the following requests would work like normal, otherwise there won't be any further requests to the locking API endpoints for that model.
- If the other endpoints (besides the `GET` endpoint) are called anyway without locking being supported, the API would then return the `HTTP 501` code because those operations are not supported. Because the endpoints shouldn't even be called in the first place on those models, this should never occur and the console error would therefore be intended.

Reply:
I thought in a similar direction last night - I think that could be a clean way to solve this (as I agree that "just working" isn't that ideal).